commit files for board

floresjann opened the pull request.

floresjann committed:
files for dietcakeboard

rgmtan commented on app/controllers/thread_controller.php:
```diff
@@ -0,0 +1,73 @@
+<?php
+
+    class ThreadController extends AppController
```
please fix indention similar to models/thread.php

rgmtan commented on app/controllers/thread_controller.php:
```diff
@@ -0,0 +1,73 @@
+        }
+    }
+
+    class ValidationException extends AppException
```
please transfer this in app_exception.php
`in app_exception.php`
```php
<?php
class AppException extends Exception
{
}

class ValidationException extends AppException
{
}
```

rgmtan commented on app/controllers/thread_controller.php:
```diff
@@ -0,0 +1,73 @@
+                    }
+                    break;
+                default:
+                    throw new NotFoundException("{$page} is not found");
```
`NotFoundException` is not declared po, please declare in app_exception.php

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,66 @@
+        $db = DB::conn();
+        $rows = $db->rows('SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC', array($this->id));
+        foreach ($rows as $row)
+            {
```
please fix indention po

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,66 @@
+    {
+        $db = DB::conn();
+        $db->query(
+            'INSERT INTO comment SET thread_id = ?, username = ?, body = ?, created = NOW()', array($this->id, $comment->username, $comment->body));
```
you can use `insert` method instead
```php
$params = array(
    'thread_id' => $this->id,
    'username' => $comment->username,
    'body' => $comment->body,
    'created' => date('Y-m-d H:i:s')
);
$db->insert('comment', $params);
```

rgmtan replied:
Also, please transfer this procedure in comment model as well since it involves the comment table. You can modify this method like this
```php
public function writeComment($body)
{
    Comment::create($body, $this);
}
```
then in `models/comment.php`
```php
public static function Comment($body, Thread $thread)
{
    $db = DB::conn();
    $params = array(
        'thread_id' => $thread->id,
        'username' => $thread->username,
        'body' => $body,
        'created' => date('Y-m-d H:i:s')
    );
    $db->insert('comment', $params);
}

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,66 @@
+        $comment->validate();
+        if ($this->hasError() || $comment->hasError()) {
+            throw new ValidationException('invalid thread or comment');
+}
```
please fix indent

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,66 @@
+    {
+        $comments = array();
+        $db = DB::conn();
+        $rows = $db->rows('SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC', array($this->id));
```
this should be done in the comment model. You can revise it like this
```php
public function getComments()
{
    return Comment::getByThreadId($this->id);
}
```
in `models/comment.php`
```php
public static function getByThreadId($thread_id)
{
    $comments = array();
    $db = DB::conn();
    $rows = $db->rows('SELECT * FROM comment WHERE thread_id = ?', array($this->id));
    foreach ($rows as $row) {
        $comments[] = new self($row);
    }
    return $comments;
}
```

floresjann committed:
transfered class ValidationException extends AppException from thread_contoller
declared NotFoundException

floresjann committed:
fixed indention

floresjann committed:
transfered getComments() from thread.php

rgmtan commented on app/models/comment.php:
```diff
@@ -0,0 +1,41 @@
+            }
+        }
+
+        public function getComments()
```
Please move this back in thread model po

floresjann committed:
fix indentions

floresjann committed:
transfered getComments() from comment
fixed indentions

floresjann committed:
fixed indentions

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,59 @@
+            return new self($row);
+        }
+
+        public function write(Comment $comment)
```
Please move this in comment model po rename to 
```php
public function write()
{
    if (!$this->validate()) {
        throw new ValidationException('wrong input');
    }

    $db = DB::conn();
    $params = array(
        'thread_id' => $this->thread_id,
        'username' => $this->username,
        'body' => $this->body,
        'created' => date('Y-m-d H:i:s'),
    );
    $db->insert('comment', $params);
}

floresjann committed:
tranfered getComments()
fixed indentions

rgmtan commented on app/controllers/thread_controller.php:
```diff
@@ -0,0 +1,69 @@
+                    $comment->username = Param::get('username');
+                    $comment->body = Param::get('body');
+                    try {
+                        $thread->write($comment);
```
Based from this [comment](https://github.com/floresjann/DietCake-Board/pull/1/files#r122123639) po, please try this
from line `20` modify :
```php
$page = Param::get('page_next', 'write');
if ($page == 'write_end') {
    $params = array(
        'thread_id' => Param::get('thread_id'),
        'username' => Param::get('username'),
        'body' => Param::get('body'),
    );
    $comment = new Comment($params);
    try {
        $comment->write();
    } catch(ValidationException $e) {
        $page = 'write';
    }
}

you can use if () instead of switch statement since you only have 2 cases.

floresjann committed:
fixed indentions
change switch ase to if for function write in thread_controller
fixed validation for comment in comment.php

rgmtan commented on app/controllers/thread_controller.php:
```diff
@@ -0,0 +1,68 @@
+    public function write()
+    {
+        $thread = Thread::get(Param::get('thread_id'));
+        $comment = new Comment;
```
please remove this line since you declared in the bottom.

rgmtan commented on app/controllers/thread_controller.php:
```diff
@@ -0,0 +1,68 @@
+        $thread = Thread::get(Param::get('thread_id'));
+        $comment = new Comment;
+        $page = Param::get('page_next', 'write');
+        $page = Param::get('page_next', 'write');
```
double declaration po, please remove the other one

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,52 @@
+        $db->query('INSERT INTO thread SET title = ?, created = NOW()', array($this->title));
+        $this->id = $db->lastInsertId();
+        // write first comment at the same time
+        $this->write($comment);
```
this will produce an error since write() method was moved in comment model. please modify to 
```php
$comment->thread_id = $this->id;
$comment->write();

rgmtan commented on app/models/comment.php:
```diff
@@ -0,0 +1,45 @@
+
+class Comment extends AppModel
+{
+
```
please remove extra space po, same with Line 2

floresjann committed:
Removed: $comment = new Comment; and  $page = Param::get('page_next', 'write');
Changed $this->write($comment) to $comment->thread_id => this->id;

rgmtan commented on app/helpers/html_helper.php:
```diff
@@ -0,0 +1,14 @@
+<?php
+
```
please remove extra space

rgmtan commented on app/controllers/thread_controller.php:
```diff
@@ -0,0 +1,66 @@
+<?php
+
```
same from this [comment](https://github.com/floresjann/DietCake-Board/pull/1/files#r122352853)

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,53 @@
+<?php
+
```
same from this [comment](https://github.com/floresjann/DietCake-Board/pull/1/files#r122352853)

rgmtan commented on app/models/thread.php:
```diff
@@ -0,0 +1,53 @@
+        $comment->write();
+        $db->commit();
+    }
+
```
same [here](https://github.com/floresjann/DietCake-Board/pull/1/files#r122352853)

floresjann committed:
fixed indentions
added usercontrollser, user view and user model

rgmtan commented on app/models/user.php:
```diff
@@ -0,0 +1,67 @@
+            $this->user_validated = false;
+        }
+        $username = $this->username;
+        $password = $this->password;
```
no need to assign the variables po since it is already a property of this class.

rgmtan commented on app/models/user.php:
```diff
@@ -0,0 +1,67 @@
+        $password = $this->password;
+
+        $db = DB::conn();
+        $user = $db->row('SELECT username, password FROM user WHERE username = ? and password = ?', array($username, md5($password)));
```
you can use the class properties as param for the query
```php
$db->row('SELECT query.....', array($this->username, md5($this->password));
```

rgmtan commented on app/models/user.php:
```diff
@@ -0,0 +1,67 @@
+
+        if (!$user) {
+            throw new RecordNotFoundException();
+            $this->user_validated = false;
```
please

rgmtan replied:
Since you are throwing the exception before assigning value to the property, the property's value will not change. Please interchange L31 and L32.

rgmtan commented on app/models/user.php:
```diff
@@ -0,0 +1,67 @@
+    {
+        if (!$this->isUserExisting() || !$this->isPasswordMatch()) {
+            $this->user_validated = false;
+        }
```
I don't think this check is needed since fetching the user data below is enough for login check.

rgmtan commented on app/controllers/user_controller.php:
```diff
@@ -0,0 +1,68 @@
+                $password = $user->password = Param::get('password');
+                $user->password = $password;
+                try {
+                    $user = $user->login($user);
```
you are passing the same object to itself as parameter which is redundant. Please remove 
```php
$user = $user->login();
```

rgmtan commented on app/controllers/user_controller.php:
```diff
@@ -0,0 +1,68 @@
+
+            case 'login_end':
+                $user->username = Param::get('username');
+                $password = $user->password = Param::get('password');
```
redundant variable declaration po. You can assign the property directly to $user object without the need of this variable.
```php
$user->password = Param::get('password');
```

rgmtan commented on app/models/user.php:
```diff
@@ -0,0 +1,67 @@
+        ),
+    );
+
+    public function login(User $user)
```
Please remove parameter User as it is not needed in this method.

rgmtan commented on app/models/user.php:
```diff
@@ -0,0 +1,67 @@
+<?php
+class User extends AppModel
+{
+    const ERR_DUPLICATE_ENTRY = 1062;
```
is this used po? if not, please remove.

rgmtan commented on app/controllers/user_controller.php:
```diff
@@ -0,0 +1,68 @@
+            case 'login':
+
+                break;
+
```
please remove extra spaces in Line 12, 40, 42 and 54

floresjann committed:
user.php
1. removed variables-> $username, $password
2. use the class properties as param for the query  array($this->username, md5($this->password));
3. removed redundant validation - removed method for (isPasswordMatch)
4. interchange exception and $user_validated
5. removed parameter for function login()
6. removed   const ERR_DUPLICATE_ENTRY = 1062;

user_controller.php
1. removed parameter for $user = $user->login();
2. change redundant declaration $user->password = Param::get('password');
3. removed extra spaces
